### PR TITLE
Update Nginx proxy docs for new proxy changes

### PR DESCRIPTION
**Pull request by derrickmehaffy**
Signed-off-by: Derrick Mehaffy <derrickmehaffy@gmail.com>

#### Description of what you did:

This updates the Nginx proxy documentation to include the new changes to the server config + the new proxy changed merged in #5833

### CODE REVIEW

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md [hunk 1]
```diff
@@ -65,6 +83,218 @@ server {
...
+
+:::: tab Sub-Domain
+
+#### Sub-Domain Strapi config
+
+---
+
+Example Domain: `api.example.com`
+
+Path: `config/server.js`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/nginx-proxy.md:243]:
```diff
- Path: `config/server.js`
+ **Path —** `config/server.js`
```

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md [hunk 2]
```diff
@@ -65,6 +83,218 @@ server {
...
+
+::: warning
+Please note that this config is not focused on the frontend hosting, you will most likely need to adjust this to your frontend software requirements, it is only being shown here as an example.
+:::
+
+---
+
+Example Domain: `example.com/api`
+
+Path: `/etc/nginx/sites-available/strapi.conf`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/nginx-proxy.md:102]:
```diff
- Path: `/etc/nginx/sites-available/strapi.conf`
+ **Path —** `/etc/nginx/sites-available/strapi.conf`
```

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md [hunk 3]
```diff
@@ -65,6 +83,218 @@ server {
...
+Please note that this config is not focused on the frontend hosting, you will most likely need to adjust this to your frontend software requirements, it is only being shown here as an example.
+:::
+
+---
+
+Example API Domain: `example.com/api`
+
+Example Admin Domain: `example.com/dashboard`
+
+Path: `/etc/nginx/sites-available/strapi.conf`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/nginx-proxy.md:166]:
```diff
- Path: `/etc/nginx/sites-available/strapi.conf`
+ **Path —** `/etc/nginx/sites-available/strapi.conf`
```

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md [hunk 4]
```diff
@@ -65,6 +83,218 @@ server {
...
+
+:::: tab Sub-Folder-Unified
+
+#### Sub-Folder Unified Strapi config
+
+---
+
+Example Domain: `example.com/api`
+
+Path: `config/server.js`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/nginx-proxy.md:263]:
```diff
- Path: `config/server.js`
+ **Path —** `config/server.js`
```

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md [hunk 5]
```diff
@@ -65,6 +83,218 @@ server {
...
+
+#### Sub-Folder Split Strapi config
+
+---
+
+Example API Domain: `example.com/api`
+
+Example Admin Domain: `example.com/dashboard`
+
+Path: `config/server.js`
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/deployment/nginx-proxy.md:285]:
```diff
- Path: `config/server.js`
+ **Path —** `config/server.js`
```

File: docs/3.0.0-beta.x/deployment/nginx-proxy.md (hunk 1 above)
Comment (derrickmehaffy) [reply docs/3.0.0-beta.x/deployment/nginx-proxy.md:243]: Absolutely I'll fix all of them in a bit

**Pull request merged**